app/models/course/survey/response.rb

```diff
@@ -18,7 +18,8 @@ def submitted?
   def submit
     self.submitted_at = Time.zone.now
     self.points_awarded = survey.base_exp
-    self.points_awarded += survey.time_bonus_exp if submitted_at <= survey.end_at
+    self.points_awarded += survey.time_bonus_exp if \
```

Comment:
Yes. Should be this way because
- When `allow_response_after_end` is unchecked in the frontend, the `time_bonus_exp` field is hidden and `time_bonus_exp` is set to 0, so staff and students will not expect any bonus points to be awarded.
- It is possible for `end_at` to be absent when `allow_response_after_end` is false.